packages/trpc/server/routers/viewer/bookings/confirm.handler.ts

```diff
@@ -116,23 +114,25 @@ export const confirmHandler = async ({ ctx, input }: ConfirmOptions) => {
     throw new TRPCError({ code: "BAD_REQUEST", message: "Booking already confirmed" });
   }
 
-  // Check for user's slot availability for current booking start and end time
+  // Check if user has already accepted/confirmed previously a booking request
+  // for the same event and same time slot
   if (confirmed) {
-    const slotsAvailable = await getAvailableSlots({
-      input: {
+    const acceptedBooking = await prisma.booking.findFirst({
```

Comment:
Don't you want to check _both_ of these conditions?  (if there is no accepted booking, we would still want to check if the slot is availalble -it could have been taken off the board due to changes in availability, right?)

Reply:
@moilejter , getAvailableSlots() checks the availability across event of a user. 
As per the latest requirement we have to avoid double confirmation of multiple requests for same time slot within a single event only (and not across different events).

And the latest user's availability is taken into consideration while selecting the slot itself.